invokeai/app/invocations/image.py

```diff
@@ -1072,3 +1072,91 @@ def invoke(self, context: InvocationContext) -> CanvasV2MaskAndCropOutput:
             width=image_dto.width,
             height=image_dto.height,
         )
+
+
+@invocation_output("crop_to_object_output")
+class CropToObjectOutput(ImageOutput):
+    offset_top: int = OutputField(description="The number of pixels cropped from the top")
+    offset_left: int = OutputField(description="The number of pixels cropped from the left")
+    offset_right: int = OutputField(description="The number of pixels cropped from the right")
+    offset_bottom: int = OutputField(description="The number of pixels cropped from the bottom")
+
+
+@invocation(
+    "crop_to_object",
+    title="Crop to Object with Margin",
+    tags=["image", "crop"],
+    category="image",
+    version="1.0.1",
+)
+class CropToObjectInvocation(BaseInvocation, WithMetadata, WithBoard):
+    """Crops an image to a specified box around the object of specified color."""
+
+    image: ImageField = InputField(description="An input mask image with black and white content")
+    margin: int = InputField(default=0, ge=0, description="The desired margin around the object, as measured in pixels")
+    object_color: Literal["white", "black"] = InputField(
```

Comment:
SAM can output multi-colored masks, right? Maybe we want this to be a `ColorField` instead, and update the mask extraction accordingly.

A future UI component could let the user click the specific mask they want, sampling its color, and then pass that into this node. So it'd be like a two-stage filter - segment, then choose the mask.